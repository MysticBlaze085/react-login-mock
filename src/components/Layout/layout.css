.wrapper {
  display: flex;
  flex-direction: column;
  flex: 1;
  height: 100vh;
}

.container {
  display: flex;
  flex: 1;
  align-items: center;
}
