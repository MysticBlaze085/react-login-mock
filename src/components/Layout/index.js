import React from "react";
import PropTypes from "prop-types";

import Header from "../Header";
import "./layout.css";

function Layout({ title = "", children }) {
  return (
    <div className="wrapper">
      <Header role="banner" title={title} />
      <main className="container">{children}</main>
    </div>
  );
}

Layout.propTypes = {
  children: PropTypes.any,
};

export default Layout;
