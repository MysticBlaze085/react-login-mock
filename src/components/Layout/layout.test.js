import React from "react";
import { render } from "@testing-library/react";
import Layout from "./index";

test("renders layout component", () => {
  const { asFragment, getByRole } = render(<Layout />);
  expect(asFragment()).toMatchSnapshot();
  expect(getByRole("banner")).toBeInTheDocument();
});
