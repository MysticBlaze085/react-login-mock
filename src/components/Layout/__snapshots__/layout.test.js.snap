// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders layout component 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <header
      class="header"
    >
      <div
        class="logo-view"
      >
        <a
          href="/"
        >
          <img
            alt="Dev World Logo"
            class="logo-header"
            src="logo.svg"
          />
        </a>
        <p
          class="title"
          data-testid="header-title"
        />
      </div>
      <menu>
        <a
          class="link"
          data-testid="nav-about"
          href="/about"
        >
          About
        </a>
        <a
          class="link"
          data-testid="nav-profile"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="link"
          data-testid="nav-logout"
          href="/login"
        >
          Logout
        </a>
      </menu>
    </header>
    <main
      class="container"
    />
  </div>
</DocumentFragment>
`;
